src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -216,7 +216,8 @@ protected virtual void Dispose(bool disposing)
             };
 
             await CreateVersionsAsync(CancellationToken.None);
-            OnCurrentPackageChanged();
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(OnCurrentPackageChanged)
+                .PostOnFailure(nameof(DetailControlModel), nameof(OnCurrentPackageChanged));
```

Comment:
Should the method name be `SetCurrentPackageAsync` instead?
We use the caller's name in many places. Is that the right approach?
Ex: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs#L205

Reply:
I think renaming anything is out of scope for this PR